bindings/python/mappings.py

```diff
@@ -228,14 +229,25 @@ def lowest_costs(self, nresults=5):
         result = []
         n = 0
         for inputs in self.input_routes:
-            owncost = 1
-            for cost, idx in get_lowest_costs(inputs, nresults=nresults):
-                if isinstance(self.cost, Callable):
-                    values = get_values(inputs, idx, magnitudes=True)
-                    owncost = self.cost(**values)
+            costs = np.array([0.0])
+            for input in inputs.values():
+                if isinstance(input, MappingStep):
+                    res = input.lowest_costs(nresults=nresults)
+                    costs = np.tile(costs, len(res)) + np.repeat(
+                        [c for c, _ in sorted(res, key=lambda x: x[1])],
+                        len(costs))
                 else:
-                    owncost = self.cost
-                result.append((cost + owncost, n + idx))
+                    costs += input.cost
+
+            res = sorted((c, i) for i, c in enumerate(costs))[:nresults]
+            if isinstance(self.cost, Callable):
```

Comment:
Can you add a comment on how costs are calculated for Callable and how this differs from MappingStep?

Reply:
Done